### PR TITLE
Turn Python linter on by default

### PR DESCRIPTION
**Pull request by szalpal**
Signed-off-by: szalpal <mszolucha@nvidia.com>

## Category:

**Refactoring** (*Redesign of existing code that doesn't affect functionality*)

## Description:

The final PR for the Python Linter effort. This PR fixes the remaining linter problems in Python code and turns Python linter on in the CI. From now on, we'll have pretty Python code :)

## Additional information:
`lint.cmake` file contains the addition to `make lint` target. The remaining files are just fixing linter errors.

### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
Linter properly fails. Next step: turn it truly on, when all the fixes are applied.

**Comment by szalpal** [comment]
!build

File: dali/test/python/test_copy_to_external_torch.py [hunk 1]
```diff
@@ -12,36 +12,39 @@
...
-    types.DALIDataType.INT32   : torch.int32,
-    types.DALIDataType.INT64   : torch.int64
+    types.DALIDataType.FLOAT: torch.float32,
+    types.DALIDataType.FLOAT64: torch.float64,
+    types.DALIDataType.FLOAT16: torch.float16,
+    types.DALIDataType.UINT8: torch.uint8,
+    types.DALIDataType.INT8: torch.int8,
+    types.DALIDataType.INT16: torch.int16,
+    types.DALIDataType.INT32: torch.int32,
+    types.DALIDataType.INT64: torch.int64
```
Comment (mzient) [inline dali/test/python/test_copy_to_external_torch.py:44]: Is that change really necessary?

File: dali/test/python/test_dali_tf_dataset_graph.py [hunk 2]
```diff
@@ -58,13 +65,8 @@ def run_tf_dataset_with_random_input(dev, max_shape, dtype, batch):
...
         get_pipeline_desc=external_source_tester(max_shape, dtype, iterator, batch=batch),
-        to_dataset=external_source_converter_with_callback(RandomSampleIterator,
-                                                           max_shape,
-                                                           dtype,
-                                                           0,
-                                                           1e10,
-                                                           min_shape,
-                                                           batch=batch))
+        to_dataset=external_source_converter_with_callback(RandomSampleIterator, max_shape, dtype,
+                                                           0, 1e10, min_shape, batch=batch))
```
Comment (mzient) [inline dali/test/python/test_dali_tf_dataset_graph.py:69]: Why?

File: dali/test/python/test_dali_tf_dataset_graph.py [hunk 3]
```diff
@@ -140,23 +135,23 @@ def test_tf_dataset_with_stop_iter():
...
-                    np.full((2, 4), -42, dtype=np.int64),
-                    np.full((3, 5), -666.0, dtype=np.float32),
-                    np.full((1, 7), 5, dtype=np.int8)
-                ]]
+start_values = [
+    [np.full((2, 4), -42, dtype=np.int64), np.full((3, 5), -123.0, dtype=np.float32)],
+    [np.full((3, 5), -3.14, dtype=np.float32)],
+    [np.full((2, 4), -42, dtype=np.int64), np.full((3, 5), -666.0, dtype=np.float32),
+     np.full((1, 7), 5, dtype=np.int8)]
+]
```
Comment (mzient) [inline dali/test/python/test_dali_tf_dataset_graph.py:154]: This looks like two autoformatters fighting.... Was the previous variant somehow offending?

File: dali/test/python/test_dali_tf_dataset_graph.py [hunk 4]
```diff
@@ -171,18 +166,22 @@ def test_tf_dataset_multi_input():
 
 
 def run_tf_with_dali_external_source(dev, es_args, ed_dev, dtype, *_):
-    run_tf_dataset_graph(dev,
+    run_tf_dataset_graph(
+        dev,
         get_pipeline_desc=get_external_source_pipe(es_args, dtype, ed_dev),
-        to_dataset=external_source_to_tf_dataset,
-        to_stop_iter=True)
+        to_dataset=external_source_to_tf_dataset, to_stop_iter=True)
```
Comment (mzient) [inline dali/test/python/test_dali_tf_dataset_graph.py:172]:
```suggestion
        to_dataset=external_source_to_tf_dataset,
        to_stop_iter=True)
```

File: dali/test/python/test_data_containers.py [hunk 5]
```diff
@@ -170,7 +182,7 @@ def define_graph(self):
         [pipe.build() for pipe in pipes]
 
         iters = pipes[0].epoch_size("Reader")
-        assert(all(pipe.epoch_size("Reader") == iters for pipe in pipes))
+        assert (all(pipe.epoch_size("Reader") == iters for pipe in pipes))
```
Comment (mzient) [inline dali/test/python/test_data_containers.py:185]:
```suggestion
        assert all(pipe.epoch_size("Reader") == iters for pipe in pipes)
```

File: dali/test/python/test_operator_numba_func_gpu.py [hunk 6]
```diff
@@ -43,32 +44,41 @@ def set_all_values_to_255_sample(in_arr, out_arr):
         out_arr[x][y] = 255
 
 
-def set_output_to_input_plus_5_sample(in_arr, out_arr):
+def set_output_to_input_plus_5_sample(inp, out):
     x, y = cuda.grid(2)
-    if x < in_arr.shape[0] and y < in_arr.shape[1] and x < out_arr.shape[0] and y < out_arr.shape[1]:
-        out_arr[x][y] = in_arr[x][y] + 5
+    if x < inp.shape[0] and y < inp.shape[1] and x < out.shape[0] and y < out.shape[1]:
```
Comment (mzient) [inline dali/test/python/test_operator_numba_func_gpu.py:49]: :+1:

File: dali/test/python/test_operator_numba_func_gpu.py [hunk 7]
```diff
@@ -43,32 +44,41 @@ def set_all_values_to_255_sample(in_arr, out_arr):
...
 @pipeline_def
-def numba_func_cuda_pipe(shapes, dtype, run_fn=None, out_types=None, in_types=None, outs_ndim=None, ins_ndim=None, blocks=None, threads_per_block=None, setup_fn=None, batch_processing=None):
-    data = fn.external_source(lambda: get_data(shapes, dtype), batch=True, device = "gpu")
-    return numba_function(data, run_fn=run_fn, out_types=out_types, in_types=in_types, outs_ndim=outs_ndim, ins_ndim=ins_ndim, blocks=blocks, threads_per_block=threads_per_block, setup_fn=setup_fn, batch_processing=batch_processing, device='gpu')
-
-
-def _testimpl_numba_func(shapes, dtype, run_fn, out_types, in_types, outs_ndim, ins_ndim, blocks, threads_per_block, setup_fn, batch_processing, expected_out):
+def numba_func_cuda_pipe(shapes, dtype, run_fn=None, out_types=None, in_types=None, outs_ndim=None,
+                         ins_ndim=None, blocks=None, threads_per_block=None, setup_fn=None,
+                         batch_processing=None):
```
Comment (mzient) [inline dali/test/python/test_operator_numba_func_gpu.py:66]:
```suggestion
def numba_func_cuda_pipe(shapes, dtype, run_fn=None,
                         out_types=None, in_types=None, outs_ndim=None, ins_ndim=None,
                         blocks=None, threads_per_block=None,
                         setup_fn=None,
                         batch_processing=None):
```
...or one per line

File: dali/test/python/test_operator_warp.py [hunk 8]
```diff
@@ -12,7 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import nose_utils
```
Comment (mzient) [inline dali/test/python/test_operator_warp.py:15]: # noqa - this import has a side effect and is required for Python 3.10

File: dali/test/python/test_operator_resize.py [hunk 9]
```diff
@@ -670,20 +671,26 @@ def test_checkerboard_dali_vs_onnx_ref():
...
+    # checkerboard, linear_coeffs_antialias, output_size=out_size)
     # np.save(ref_resized_linear_antialias_filename, ref_resized_linear_antialias)
-    # ref_resized_cubic = interpolate_nd(checkerboard, lambda x, _: cubic_coeffs(x, A=-0.5), output_size=out_size)
+    # ref_resized_cubic = interpolate_nd(
+    # checkerboard, lambda x, _: cubic_coeffs(x, A=-0.5), output_size=out_size)
     # np.save(ref_resized_cubic_filename, ref_resized_cubic)
-    # ref_resized_cubic_antialias = interpolate_nd(checkerboard, lambda x, scale: cubic_coeffs_antialias(x, scale, A=-0.5), output_size=out_size)
+    # ref_resized_cubic_antialias = interpolate_nd(
+    # checkerboard, lambda x, scale: cubic_coeffs_antialias(x, scale, A=-0.5), output_size=out_size)
     # np.save(ref_resized_cubic_antialias_filename, ref_resized_cubic_antialias)
```
Comment (mzient) [inline dali/test/python/test_operator_resize.py:694]: Oh, no! This was _code_ inside - now it's not valid.

Comment (szalpal) [reply dali/test/python/test_operator_resize.py:694]: The code violated the linter, therefore I've broken it into separate lines. I have no way of testing the code, best I can do is to break it better or ask author (@jantonguirao) to give some guidelines. This is rather fresh code, so it shouldn't be a problem

File: dali/test/python/test_dali_tf_dataset_graph.py (hunk 2 above)
Comment (szalpal) [reply dali/test/python/test_dali_tf_dataset_graph.py:69]: I believe this enhances readability - this way it's more obvious that these arguments are not to the `run_tf_dataset_graph` function, but the args to the `external_source_converter_with_callback` function.

File: dali/test/python/test_dali_tf_dataset_graph.py (hunk 3 above)
Comment (szalpal) [reply dali/test/python/test_dali_tf_dataset_graph.py:154]: The formatting was inconsistent. First two elements of `start_values` begun in the same line as the opening bracket `[`, while the last one had newline inserted between. I've made it consistent.
```python
start_values = [[np.full((2, 4), -42, dtype=np.int64),
                 np.full((3, 5), -123.0, dtype=np.float32)],
                [np.full((3, 5), -3.14, dtype=np.float32)],
                [
                    np.full((2, 4), -42, dtype=np.int64),
                    np.full((3, 5), -666.0, dtype=np.float32),
                    np.full((1, 7), 5, dtype=np.int8)
                ]]
```

File: dali/test/python/test_operator_resize.py (hunk 9 above)
Comment (szalpal) [reply dali/test/python/test_operator_resize.py:694]: The proposal with "prettier" formatting (although I like the one in the PR better)
```python
    # from onnx.backend.test.case.node.resize import interpolate_nd, linear_coeffs, \
    #     linear_coeffs_antialias, cubic_coeffs, cubic_coeffs_antialias
    # 
    # ref_resized_linear = interpolate_nd(checkerboard, lambda x, _: linear_coeffs(x),
    #                                     output_size=out_size)
    # np.save(ref_resized_linear_filename, ref_resized_linear)
    # ref_resized_linear_antialias = interpolate_nd(checkerboard, linear_coeffs_antialias,
    #                                               output_size=out_size)
    # np.save(ref_resized_linear_antialias_filename, ref_resized_linear_antialias)
    # ref_resized_cubic = interpolate_nd(checkerboard, lambda x, _: cubic_coeffs(x, A=-0.5),
    #                                    output_size=out_size)
    # np.save(ref_resized_cubic_filename, ref_resized_cubic)
    # ref_resized_cubic_antialias = interpolate_nd(checkerboard,
    #                                              lambda x, scale: cubic_coeffs_antialias(x, scale,
    #                                                                                      A=-0.5),
    #                                              output_size=out_size)
    # np.save(ref_resized_cubic_antialias_filename, ref_resized_cubic_antialias)
```

Comment (jantonguirao) [reply dali/test/python/test_operator_resize.py:694]: I prefer your second proposal. It should look like code, just commented out. You can easily uncomment, autoformat and comment back, if you wish

Comment (szalpal) [reply dali/test/python/test_operator_resize.py]: Applied the second one

File: dali/test/python/test_operator_warp.py (hunk 8 above)
Comment (szalpal) [reply dali/test/python/test_operator_warp.py:15]: Done

File: dali/test/python/test_copy_to_external_torch.py [hunk 10]
```diff
@@ -63,14 +66,13 @@ def feed_ndarray(tensor_or_tl, arr, cuda_stream=None, non_blocking=False):
     else:
         dali_tensor = tensor_or_tl
 
-
     assert dali_type == arr.dtype, ("The element type of DALI Tensor/TensorList"
-            " doesn't match the element type of the target PyTorch Tensor:"
-            "{} vs {}".format(dali_type, arr.dtype))
+                                    " doesn't match the element type of the target PyTorch Tensor:"
+                                    f"{dali_type} vs {arr.dtype}")
```
[the remark this answers is not in the document]
Comment (szalpal) [reply dali/test/python/test_copy_to_external_torch.py]: Done

File: dali/test/python/test_copy_to_external_torch.py (hunk 1 above)
Comment (szalpal) [reply dali/test/python/test_copy_to_external_torch.py:44]: Yes, I believe the linter has complained here

File: dali/test/python/test_dali_tf_dataset_graph.py (hunk 2 above)
Comment (mzient) [reply dali/test/python/test_dali_tf_dataset_graph.py:69]: Fair enough.

**Review by mzient — APPROVED** [review]

**Review by awolant — APPROVED** [review]

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
L1 tests passed
